src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2728,39 +2729,48 @@ void java_bytecode_convert_method(
   java_bytecode_convert_method(class_symbol, method);
 }
 
-const bool java_bytecode_convert_methodt::is_method_inherited(
+bool java_bytecode_convert_methodt::is_method_inherited(
   const irep_idt &classname, const irep_idt &methodid) const
 {
-  class_hierarchyt ch;
-  namespacet ns(symbol_table);
-  ch(symbol_table);
+  resolve_concrete_function_callt call_resolver(symbol_table);
+  const resolve_concrete_function_callt ::concrete_function_callt &
+    resolved_call=call_resolver(classname, methodid);
 
-  std::string stripped_methodid(id2string(methodid));
-  stripped_methodid.erase(0, classname.size());
-
-  const std::string &classpackage=java_class_to_package(id2string(classname));
-  const auto &parents=ch.get_parents_trans(classname);
-  for(const auto &parent : parents)
+  if(resolved_call.is_valid())
   {
-    const irep_idt id=id2string(parent)+stripped_methodid;
-    const symbolt *symbol;
-    if(!ns.lookup(id, symbol) &&
-       symbol->type.id()==ID_code)
-    {
-      const auto &access=symbol->type.get(ID_access);
-      if(access==ID_public || access==ID_protected)
-        return true;
-      // methods with the default access modifier are only
-      // accessible within the same package.
-      else if(access==ID_default &&
-              java_class_to_package(id2string(parent))==classpackage)
-        return true;
-      else if(access==ID_private)
-        continue;
-      else
-        INVARIANT(false, "Unexpected access modifier.");
+    const symbolt &function_symbol=
+      symbol_table.lookup(resolved_call.get_virtual_method_name());
+
+    INVARIANT(function_symbol.type.id()==ID_code, "Function must be code");
+
+    const auto &access=function_symbol.type.get(ID_access);
+    if(access==ID_public || access==ID_protected)
+    {
+      return true;
     }
+    // methods with the default access modifier are only
+    // accessible within the same package.
+
+    else if(access==ID_default)
```

Comment:
That + the fact that all the cases are basic blocks ending in a return convinced me.